### PR TITLE
Feature/538 surrounding cyan waterbodies

### PR DESCRIPTION
**Pull request by maxdiebold-erg**
## Related Issues:
* [HMW-538](https://jira.epa.gov/browse/HMW-538)

## Main Changes:
* Added the CyAN layer to the Surrounding Features widget.
* Added the CyAN layer to the legend.

## Steps To Test:
1. Navigate to http://localhost:3000/community/lake%20okeechobee/monitoring.
2. Toggle on the _CyAN Waterbodies_ layer in the Surrounding Features widget.
3. Pan to a cyan waterbody outside the HUC boundaries, and click it to open a popup.
4. Verify that all features in the popup work as expected (especially the satellite imagery).
5. Click the waterbody inside the HUC boundaries.
6. Verify the satellite imagery is cleared from the previous waterbody.
7. Confirm the legend is updated when a cyan layer is visible, and that the symbol is not duplicated when both cyan layers are visible.

### CODE REVIEW

File: app/client/src/utils/hooks/cyanWaterbodies.ts [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+import SimpleFillSymbol from '@arcgis/core/symbols/SimpleFillSymbol';
+import SimpleRenderer from '@arcgis/core/renderers/SimpleRenderer';
+import { useCallback, useContext, useEffect, useRef, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+// contexts
+import { useFetchedDataDispatch } from 'contexts/FetchedData';
+import { LocationSearchContext } from 'contexts/locationSearch';
+import { useServicesContext } from 'contexts/LookupFiles';
+// utils
+import {
```
Comment (cschwinderg) [inline app/client/src/utils/hooks/cyanWaterbodies.ts:17]:
```suggestion
// utils
import { useDynamicPopup } from 'utils/hooks';
import {
```

This is related to getting the `Change to this location` section in the surrounding features popups. 

This and other changes below will need to be done in the `dischargers.ts`, `monitoringLocations.tsx`, and `streamgages.ts` files.

File: app/client/src/utils/hooks/cyanWaterbodies.ts [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+  const { getMapView } = useContext(LocationSearchContext);
+  const services = useServicesContext();
+  const navigate = useNavigate();
+
+  const buildBaseLayer = useCallback(
+    (type: SublayerType) => {
+      return buildLayer(navigate, services, type, getMapView);
+    },
+    [getMapView, navigate, services],
+  );
```
Comment (cschwinderg) [inline app/client/src/utils/hooks/cyanWaterbodies.ts:57]:
```suggestion
export function useCyanWaterbodiesLayers() {
  // Build the base feature layer
  const { getMapView } = useContext(LocationSearchContext);
  const { getTitle, getTemplate } = useDynamicPopup();
  const services = useServicesContext();
  const navigate = useNavigate();

  const buildBaseLayer = useCallback(
    (type: SublayerType) => {
      return buildLayer(navigate, services, type, getMapView, getTitle, getTemplate);
    },
    [getMapView, getTemplate, getTitle, navigate, services],
  );
```

This is related to getting the Change to this location section in the surrounding features popups.

File: app/client/src/utils/hooks/cyanWaterbodies.ts [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+    });
+  });
+}
+
+function buildLayer(
+  navigate: NavigateFunction,
+  services: ServicesState,
+  type: SublayerType,
+  getMapView: () => __esri.MapView,
+) {
```
Comment (cschwinderg) [inline app/client/src/utils/hooks/cyanWaterbodies.ts:163]:
```suggestion
function buildLayer(
  navigate: NavigateFunction,
  services: ServicesState,
  type: SublayerType,
  getMapView: () => __esri.MapView,
  getTitle: Function,
  getTemplate: Function,
) {
```

This is related to getting the Change to this location section in the surrounding features popups.

File: app/client/src/utils/hooks/cyanWaterbodies.ts [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+        getPopupTitle(feature.graphic.attributes),
+      content: (feature: __esri.Feature) =>
+        getPopupContent({
+          feature: feature.graphic,
+          mapView: getMapView(),
+          navigate,
+          services,
+        }),
+      outFields: ['*'],
+    },
```
Comment (cschwinderg) [inline app/client/src/utils/hooks/cyanWaterbodies.ts:200]:
```suggestion
    popupTemplate: {
      title: getTitle,
      content: getTemplate,
      outFields: ['*'],
    },
```

This is related to getting the Change to this location section in the surrounding features popups.

**Review by cschwinderg — CHANGES_REQUESTED** [review]
Works well!

I did notice that the `Change to this location` section only works for the All Waterbodies layer it should work for all surrounding layers. This has probably been an issue for a while and I just didn't notice it. Most of the code suggestions below are related to this. I only put in the suggestions for the `cyanWaterbodies.ts` file, but these changes will need to be implemented in other surrounding layer files. I tested these changes and they seem to work, but feel free to implement differently.  

Section is visible for All Waterbodies.
image

Section is not visible for CyAN.
image

File: app/cypress.log [hunk 5]
```diff
@@ -0,0 +1,196 @@
+
```
[the remark this answers is not in the document]
Comment (maxdiebold-erg) [reply app/cypress.log:1]: No it was not, and that was just a "manual" log file, so I'll just remove it.

File: app/client/src/utils/hooks/cyanWaterbodies.ts (hunk 1 above)
Comment (maxdiebold-erg) [reply app/client/src/utils/hooks/cyanWaterbodies.ts:17]: I knew something was off with those popups! Nice catch, I'll make those changes for all surrounding layers.

**Pull request merged**